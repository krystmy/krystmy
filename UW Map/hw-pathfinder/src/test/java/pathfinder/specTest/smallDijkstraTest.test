#test finding a small least cost path

CreateGraph graph

AddNode graph A
AddNode graph B
AddNode graph C
AddNode graph D

AddEdge graph A B 1.0
AddEdge graph B C 2.0
AddEdge graph C D 1.0
AddEdge graph A D 10.0

FindPath graph A D