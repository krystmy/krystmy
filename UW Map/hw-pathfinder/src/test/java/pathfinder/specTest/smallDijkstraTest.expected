#test finding a small least cost path

created graph graph

added node A to graph
added node B to graph
added node C to graph
added node D to graph

added edge 1.000 from A to B in graph
added edge 2.000 from B to C in graph
added edge 1.000 from C to D in graph
added edge 10.000 from A to D in graph

path from A to D:
A to B with weight 1.000
B to C with weight 2.000
C to D with weight 1.000
total cost: 4.000